src/Services/ReorderElements.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace DNADesign\Elemental\Services;
+
+use InvalidArgumentException;
+use SilverStripe\Core\Convert;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\Queries\SQLUpdate;
+use DNADesign\Elemental\Models\BaseElement;
+
+class ReorderElements
+{
+    /**
+     * @var BaseElement $block
+     */
+    private $element;
+
+    public function __construct(BaseElement $element)
+    {
+        if (!is_subclass_of($element, BaseElement::class)) {
+            throw new InvalidArgumentException(sprintf(
+                'Invalid ' . BaseElement::class . ' passed to ' . __CLASS__
+            ));
+        }
+
+        if (!$element->canEdit()) {
```

Comment:
We're missing the context here, this will have problems with using non-session based authentication since the member is always passed in the `$context` args

Reply:
Ah cool, didn't realise ta